File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
import sys
from pprint import pprint
# sys.stdin = open('input2.txt')
# 회의실 수
n = int(sys.stdin.readline().strip())
meeting_list = [list(map(int, sys.stdin.readline().strip().split())) for _ in range(n)]

# 끝나는 시간 오름차순 / 시작시간 오름차순 순으로 정렬
meeting_list.sort(key=lambda x:(x[1], x[0]))

# pprint(meeting_list)
# 회의 순서대로 넣을 stack

cnt = 1
# 회의 끝나는 시간
end = meeting_list[0][1]
for i in range(1,n):
    # 끝나는 시간은 시작시간보다 같거나 작아야함
    if meeting_list[i][0] >= end:
        cnt+=1
        end = meeting_list[i][1]

pprint(cnt)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let total = Number(input[0]);
let re = [];
for (let i = 1; i <= total; i++) {
  re.push(input[i].split(" ").map(Number));
}
re.sort((a, b) => { // 종류시점 비교, 시작시점 비교
  if(a[1] !== b[1]) return a[1]-b[1];
  else return a[0]-b[0]
});


let cnt = 1;
let cur = 0; // 첫번재 회의부터
for (let i = 1; i < re.length; i++) {
  if(re[cur][1] <= re[i][0]) {

    cur = i; // 현재회의 끝나고 다음회의
    cnt +=1;
  }
}

console.log(cnt);